### PR TITLE
feat(cli): aws compliance search command

### PR DESCRIPTION
**Pull request by dmurray-lacework**
## Summary
Add new ` compliance aws search <resource_arn>` command

```
❯ lacework comp aws search arn:aws:s3:::my-bucket
  RECOMMENDATION ID    ACCOUNT ID                REASON               SEVERITY      STATUS     
--------------------+--------------+--------------------------------+----------+---------------
  LW_S3_12            123456789   S3 bucket does not have MFA      Medium     NonCompliant  
                                     delete enabled                 
```                                                             

Returns all violations for a given resource.

## How did you test this change?

Run `lacework comp aws search <arn>`

## Issue
https://lacework.atlassian.net/browse/ALLY-1108

### CODE REVIEW

File: api/compliance_evaluations.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+//			  Filters: []api.Filter{{
+//				  Expression: "eq",
+//				  Field:      "resource",
+//				  Value:      arn:aws:s3:::my-bucket,
+//			  }},
+//		  },
+//		  Dataset: api.AwsComplianceEvaluationDataset,
+//	  }
+//  )
+//   lacework.V2.ComplianceEvaluation.Search(response, filters)
```
[the remark this answers is not in the document]
Comment (afiune) [reply api/compliance_evaluations.go:46]: should the user pass a pointer to the response?

File: api/compliance_evaluations_aws.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	return &r.Paging
+}
+func (r *ComplianceEvaluationAwsResponse) ResetPaging() {
+	r.Paging = V2Pagination{}
+}
+
+type ComplianceEvaluationAws struct {
+	Account struct {
+		AccountId    string `json:"AccountId"`
+		AccountAlias string `json:"Account_Alias"`
```
Comment (afiune) [inline api/compliance_evaluations_aws.go:38]: `Account_Alias` is that correct? 🤦🏽‍♂️

File: api/inventory.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+//			  Filters: []api.Filter{{
+//				  Expression: "eq",
+//				  Field:      "urn",
+//				  Value:      arn:aws:s3:::my-bucket,
+//			  }},
+//		  },
+//		  Dataset: api.AwsComplianceEvaluationDataset,
+//	  }
+//  )
+//   lacework.V2.Inventory.Search(response, filters)
```
Comment (afiune) [inline api/inventory.go:48]: same 
```suggestion
//   lacework.V2.Inventory.Search(& awsInventorySearchResponse, filters)
```

File: api/inventory.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+//
+func (svc *InventoryService) Search(response interface{}, filters InventorySearch) error {
+	return svc.client.RequestEncoderDecoder("POST", apiV2InventorySearch, filters, response)
+}
+
+type InventorySearch struct {
+	SearchFilter
+	Csp     inventoryType    `json:"csp"`
+	Dataset inventoryDataset `json:"dataset"`
+}
```
Comment (afiune) [inline api/inventory.go:58]: We need to write unit tests for all of these new functions in the `api` go package.

File: cli/cmd/compliance_aws.go [hunk 5]
```diff
@@ -411,13 +411,85 @@ The output from status with the --json flag can be used in the body of PATCH api
...
+				}
+			)
+			err := cli.LwApi.V2.ComplianceEvaluations.Search(&awsComplianceEvaluationSearchResponse, filter)
+			cli.StopProgress()
+			if err != nil {
+				return err
+			}
+
+			if len(awsComplianceEvaluationSearchResponse.Data) == 0 {
+				cli.OutputHuman("No results found.\n")
```
Comment (afiune) [inline cli/cmd/compliance_aws.go:449]: I don't think results is the right work here. what is the user searching for? could resources? 🤔

File: cli/cmd/compliance_aws.go [hunk 6]
```diff
@@ -411,13 +411,85 @@ The output from status with the --json flag can be used in the body of PATCH api
...
 	}
+
+	// complianceAwsListAccountsCmd represents the list-accounts inside the aws command
+	complianceAwsSearchCmd = &cobra.Command{
+		Use:   "search <resource_arn>",
+		Short: "Search for all known violations of a given resource arn",
+		Long:  `Search for all known violations of a given resource arn.`,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) error {
+			cli.StartProgress(fmt.Sprintf(" searching accounts for resource %s...", args[0]))
```
Comment (afiune) [inline cli/cmd/compliance_aws.go:422]:
```suggestion
			cli.StartProgress(fmt.Sprintf("Searching accounts for resource %s...", args[0]))
```

File: integration/compliance_aws_test.go [hunk 7]
```diff
@@ -158,3 +158,12 @@ func TestComplianceAwsGetReportRecommendationID(t *testing.T) {
...
 }
+
+func TestComplianceAwsSearch(t *testing.T) {
+	out, err, exitcode := LaceworkCLIWithTOMLConfig(
+		"compliance", "aws", "search", "example",
+	)
+	assert.Empty(t, err.String(), "STDERR should be empty")
+	assert.Equal(t, 0, exitcode, "EXITCODE is not the expected one")
+	assert.Contains(t, out.String(), "No results found.", "STDOUT changed, please check")
+}
```
Comment (afiune) [inline integration/compliance_aws_test.go:169]: can we add a successful integration test? that is, search for a dummy resource that actually exists.

Comment (afiune) [reply integration/compliance_aws_test.go:169]: maybe a silly S3 bucket

**Review by afiune — CHANGES_REQUESTED** [review]
I only see 2 possible outputs, when the resource is not found and when it is found with violations, have you tested the third one? resource found without violations.

File: api/compliance_evaluations_aws.go (hunk 2 above)
Comment (dmurray-lacework) [reply api/compliance_evaluations_aws.go:38]: Yes it is

File: cli/cmd/compliance_aws.go (hunk 5 above)
Comment (dmurray-lacework) [reply cli/cmd/compliance_aws.go:449]: It's meant to be a  "generic" response

With `ComplianceEvaluations/search ` we can't distinguish between the resource not existing and not having violations. Both return 204 nocontent

File: api/compliance_evaluations.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmurray-lacework) [reply api/compliance_evaluations.go:46]: We do use this pattern in Entities and AgentInfo search

File: cli/cmd/compliance_aws.go (hunk 6 above)
Comment (dmurray-lacework) [reply cli/cmd/compliance_aws.go:422]: capital, but we do want to include the space for the spinner

File: integration/compliance_aws_test.go (hunk 7 above)
Comment (dmurray-lacework) [reply integration/compliance_aws_test.go:169]: yes!

File: cli/cmd/compliance_aws.go [hunk 8]
```diff
@@ -411,13 +411,117 @@ The output from status with the --json flag can be used in the body of PATCH api
...
+					},
+					Dataset: api.AwsInventoryDataset,
+					Csp:     api.AwsInventoryType,
+				}
+			)
+			err := cli.LwApi.V2.Inventory.Search(&awsInventorySearchResponse, filter)
+			cli.StopProgress()
+
+			if len(awsInventorySearchResponse.Data) == 0 {
+				cli.OutputHuman("Resource %s not found. To learn how to configure Lacework with Aws visit "+
```
Comment (afiune) [inline cli/cmd/compliance_aws.go:447]:
```suggestion
				cli.OutputHuman("Resource '%s' not found.\n\nTo learn how to configure Lacework with AWS visit "+
```

File: cli/cmd/compliance_aws.go [hunk 9]
```diff
@@ -411,13 +411,117 @@ The output from status with the --json flag can be used in the body of PATCH api
...
+					"https://docs.lacework.com/onboarding/category/integrate-lacework-with-aws \n", args[0])
+				return nil
+			}
+			cli.StopProgress()
+			if err != nil {
+				return err
+			}
+
+			cli.StartProgress(fmt.Sprintf(" Searching for compliance violations for %s...", args[0]))
```
Comment (afiune) [inline cli/cmd/compliance_aws.go:456]:
```suggestion
			cli.StartProgress(fmt.Sprintf("Searching for compliance violations for '%s'...", args[0]))
```

**Review by afiune — APPROVED** [review]

**Pull request merged**